### ISSUE TITLE
Test runner feature request: capture stderr

### ISSUE THREAD

**Issue by RReverser** [emscripten-core/emscripten#18308]
Many test runners, including unittest (via `--buffer` option), allow to capture stderr and show it only as part of the failure in the end.

It would be great to implement this option in the parallel test runner, as it would allow to see all the relevant error information in the end of the runner logs (e.g. on CI) instead of hunting for the stderr output somewhere in the middle of the log. cc @sbc100

**Comment by sbc100** [comment]
Agreed!

**Comment by dschuff** [comment]
From a twitter conversation I'm having with @RReverser right now (:smile:) we might consider migrating to `pytest`, which supports stderr capture, parallel running, and xml output (which could replace #18375)

**Comment by sbc100** [comment]
I've tried and failed to replace our running in the past.  The one I thought most promising was https://github.com/CleanCut/green.  I've never found one that did test-level-parallelism, only suite-level.  But @RReverser said he might have found one that does, so maybe it do-able.

**Comment by RReverser** [comment]
> But @RReverser said he might have found one that does, so maybe it do-able.

Yeah I mentioned it in the other issue, but so that it doesn't get lost, it was python-xdist. They seem to have arbitrary parallelism, default being test-level that we want.
